python/lsst/pex/config/configField.py

```diff
@@ -94,7 +96,7 @@ def __init__(self, doc, dtype, default=None, check=None, deprecated=None):
             deprecated=deprecated,
         )
 
-    def __get__(self, instance, owner=None):
+    def __get__(self, instance, owner=None) -> FieldTypeVar:
```

Comment:
This should be annotated as an overload, too, or anyone who access the field on the type will be told their typing is wrong when it isn't.  I realize that's rare, but I think the bar for adding annotations is that if you add them they can't be lies.